elements/research/ddiotune.cc

```diff
@@ -0,0 +1,385 @@
+/*
+ * DDIOTune.{cc,hh} --
+ * Alireza Farshin
+ *
+ * Copyright (c) 2020 KTH Royal Institute of Technology
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, subject to the conditions
+ * listed in the Click LICENSE file. These conditions include: you must
+ * preserve this copyright notice, and you cannot mention the copyright
+ * holders in advertising related to the Software without their permission.
+ * The Software is provided WITHOUT ANY WARRANTY, EXPRESS OR IMPLIED. This
+ * notice is a summary of the Click LICENSE file; the license in that file is
+ * legally binding.
+ */
+
+#include <click/config.h>
+#include <click/args.hh>
+#include <click/error.hh>
+#include <cpuid.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <cmath>
+#include "ddiotune.hh"
+#include <rte_lcore.h>
+
+CLICK_DECLS
+
+/* Read from an MSR register */
+int32_t
+DDIOTune::rdmsr(int msr, uint64_t *val, int cpu)
+{
+	int fd;
+	int ret;
+	char msr_file[50];
+	sprintf(msr_file,"/dev/cpu/%d/msr",cpu);
```

Comment:
Make sure you leave a space after each argument (e.g., msr_file, "..", cpu)

Reply:
I will fix them. Forget to run reindent ;)

Reply:
Shouldn't `click_chatter` be preferred over `sprintf` ?